cluster/images/conformance/conformance-e2e.yaml

```diff
@@ -59,7 +59,13 @@ spec:
     imagePullPolicy: IfNotPresent
     env:
     - name: E2E_FOCUS
-      value: "Pods should be submitted and removed"
+      value: "[Conformance]"
+    - name: E2E_SKIP
+      value: ""
+    - name: E2E_PROVIDER
+      value: "local"
+    - name: E2E_PARALLEL
+      value: "y"
```

Comment:
The current Dockerfile used a `1` for this arg. Setting to a `y` would mean we'd pass `--nodes=25`. Is that desired for the default?

Reply:
@stevesloka yes, i'd like to try this as the default (at least for the CI jobs). sonobuoy can still use the existing defaults.

Reply:
I don't think we can do that seeing as there are `[Serial]` tests

Reply:
also at least on kind, tentative CI results suggest that running the non-serial tests in parallel is indeed flakier (and we've been running both for some time now)

https://testgrid.k8s.io/conformance-kind#kind,%20master%20(dev)%20%5Bnon-serial%5D